rust/Dockerfile

```diff
@@ -125,16 +125,21 @@ ENV PACKAGE=${PACKAGE}
 CMD $PACKAGE
 
 # used for local development
-FROM runtime as development
+FROM runtime as test
 
 RUN set -xe \
   && apk add --no-cache iperf3 bind-tools iproute2 jq
 
+# used for local development
+FROM test AS dev
+ARG TARGET
+COPY --from=builder /build/target/${TARGET}/debug/${PACKAGE} .
```

Comment:
This was to revert to how it worked before due to @thomaseizinger [feedback](https://firezonehq.slack.com/archives/C04HRQTFY0Z/p1710310997243179)

Reply:
Ah I see, by updating that I will fix this. So the preferred method is to use binaries built outside Docker or inside Docker for `dev`?

Reply:
This is the only place we build the binary inside the Docker image. On CI we build with `cross`, so if this can be removed then we can greatly simplify `rust/Dockerfile` to be just a vessel for the host-built binary.

Reply:
I think this works how it did before now. `rust/docker-compose-dev.yml` uses a host binary, and building by default with that file builds the binary inside, like it worked before.

Reply:
> Ah I see, by updating that I will fix this. So the preferred method is to use binaries built outside Docker or inside Docker for dev?

Outside of docker for dev we have `debug` for that inside the dockerfile.

That way developers with rust toolchain locally can opt-in their into building their own binary.

Reply:
What I mean then, is that  `docker-compose-dev` should use `test` then

Reply:
I built them inside the image because it meant I could do it in one command and didn't have to copy around binaries.

My requirement is that I can build images in one command that reflect my local source code. If it can automatically grab them from `target/`, even better :)

Reply:
Kk I think this works as it did before now. I'll leave the Docker build config in the Dockerfile for non-Rust devs.